staging/src/k8s.io/client-go/applyconfigurations/utils.go

```diff
@@ -1715,3 +1721,370 @@ func ForKind(kind schema.GroupVersionKind) interface{} {
 	}
 	return nil
 }
+
+type TypeConverter struct{}
+
+func (tc TypeConverter) ObjectToTyped(obj runtime.Object, opts ...typed.ValidationOptions) (*typed.TypedValue, error) {
+	gvk := obj.GetObjectKind().GroupVersionKind()
+	t := internal.Parser().Type(openAPIName(gvk))
+	switch o := obj.(type) {
+	case *unstructured.Unstructured:
+		return t.FromUnstructured(o.UnstructuredContent(), opts...)
+	default:
+		return t.FromStructured(obj, opts...)
+	}
+}
+
+func (tc TypeConverter) TypedToObject(value *typed.TypedValue) (runtime.Object, error) {
+	vu := value.AsValue().Unstructured()
+	switch o := vu.(type) {
+	case map[string]interface{}:
+		return &unstructured.Unstructured{Object: o}, nil
+	default:
+		return nil, fmt.Errorf("failed to convert value to unstructured for type %T", vu)
+	}
+}
+
+func openAPIName(kind schema.GroupVersionKind) string {
```

Comment:
Assuming theres a scheme we have access to, I think an alternative way of getting the openAPIName from the GVK at runtime using the scheme could be:

```go
namer := openapi.NewDefinitionNamer(GeneratedPackage.Scheme)
example, err := GeneratedPackage.Scheme.New(gvk)
rtype := reflect.TypeOf(example).Elem()
name, exts := namer.GetDefinitionName(rtype.PkgPath() + "." + rtype.Name())
```

Reply:
The scheme is available.  Calling `openapi.NewDefinitionNamer` directly won't work because it's packaged in `apiserver` and we don't want to depend on the server from the client, but I think we need very little from that code, so I'll take the "a little copy-paste is better than a lot of dependency" approach.

Reply:
Fixed